express-crud/server.ts

```diff
@@ -0,0 +1,141 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib';
+
+const db = new pg.Pool({
+  connectionString: 'postgress://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    //  Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades";
+      `;
+    const result = await db.query(sql);
+    const grades = result.rows;
+    res.json(grades);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (!Number.isInteger(+gradeId)) {
+      throw new ClientError(400, `need valid gradeId`);
+    }
+    const sql = `
+      select *
+        from "grades"
+        where "gradeId" = $1;
+      `;
+    const result = await db.query(sql, [gradeId]);
+    const [grades] = result.rows;
+    if (!grades) {
+      throw new ClientError(404, `${gradeId} not found`);
+    }
+    res.json(grades);
+  } catch (err) {
+    next(err);
+  }
+});
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    if (!Number.isInteger(+score)) {
+      throw new ClientError(400, `need valid score`);
+    }
+    if (score < 0 || score > 100) {
+      throw new ClientError(400, `score must be between 0-100`);
+    }
+    if (!name || !course) {
+      throw new ClientError(400, `name and course is needed`);
+    }
+    const sql = `
+      insert into "grades" ("name","course","score")
+      values($1,$2,$3)
+        returning *;
+      `;
+    const result = await db.query(sql, [name, course, score]);
+    const [addedResult] = result.rows;
+    if (!addedResult) {
```

Comment:
This will never happen on an insert. If there is a problem inserting, it will throw an exception. So you can remove this check.